src/javascript/app/pages/user/account/settings/account_closure.js

```diff
@@ -0,0 +1,93 @@
+const BinarySocket         = require('../../../../base/socket');
+const localize             = require('../../../../../_common/localize').localize;
+const Url                  = require('../../../../../_common/url');
+
+const AccountClosure = (() => {
+    const form_selector = '#form_closure';
+    let $txt_other_reason,
+        $closure_loading,
+        $closure_container,
+        $success_msg,
+        $error_msg;
+
+    const onLoad = () => {
+        $txt_other_reason  = $('#other_reason');
+        $closure_loading   = $('#closure_loading');
+        $closure_container = $('#closure_container');
+        $success_msg       = $('#msg_main');
+        $error_msg         = $('#msg_form');
+
+        $(form_selector).on('submit', (event) => {
+            event.preventDefault();
+            submitForm();
+        });
+        $closure_container.setVisibility(1);
+        $txt_other_reason.on('keyup', () => $txt_other_reason.removeClass('error-field'));
+    };
+
+    const submitForm = () => {
+        const $btn_submit = $(`${form_selector} #btn_submit`);
+        const reason = getReason();
+        if (reason) {
+            $closure_loading.setVisibility(1);
+            $btn_submit.attr('disabled', true);
+
+            const data  = { account_closure: 1, reason };
+            BinarySocket.send(data).then((response) => {
+                if (response.error) {
+                    showFormMessage(response.error.message || localize('Sorry, an error occurred while processing your request.'));
+                    $btn_submit.attr('disabled', false);
+                } else {
+                    $closure_loading.setVisibility(0);
+                    $closure_container.setVisibility(0);
+                    $success_msg.setVisibility(1);
+
+                    setTimeout(() => window.location.href = Url.urlFor('home'), 10000);
+                }
+            });
+        } else {
+            setTimeout(() => { $btn_submit.removeAttr('disabled'); }, 1000);
+        }
+    };
+
+    const showFormMessage = (localized_msg) => {
+        $.scrollTo($('#reason'), 500, { offset: -20 });
+        $error_msg
+            .attr('class', 'errorfield')
```

Comment:
we are still using both `error-field` and `errorfield` in this PR, are they supposed to be the same or different?

Reply:
They're slightly different. `error-field` has a `border-color: red` attribute, which we don't need, we just want a `color: red`